server/service/src/sync/sync_on_central/mod.rs

```diff
@@ -126,13 +149,63 @@ pub async fn push(
         repo.upsert_one(&buffer_row)?;
     }
 
-    // TODO we need to trigger integrate records for just 1 site?
-    // See issue: https://github.com/msupply-foundation/open-msupply/issues/3294
-    if total_records <= records_in_this_batch {
-        service_provider.sync_trigger.trigger();
+    if is_last_batch {
+        set_integrating(response.site_id, true);
+
+        integrate_and_translate_sync_buffer(&ctx.connection, true, None, Some(response.site_id))
+            .await
+            // TODO map to IntegrationError once implemented
+            // .map_err(Error::IntegrationError)?;
+            .map_err(|e| {
+                Error::OtherServerError(
+                    "Error integrating records: ".to_string() + e.to_string().as_str(),
```

Comment:
Was hoping to be able to respond with a more explicit IntegrationError here if something goes wrong in that integration step, but looks like I'd need to implement Serialize/Deserialize for the anyhow Error... which sounded complicated?

If there's actually a simple way to do that I would love to know! Otherwise went with just `OtherServerError` here...

Reply:
Actually also... currently the `wait_for_integration` after calling push doesn't do anything, because because I'm awaiting integration here... maybe I shouldn't be?

Before we were just triggering sync - but then we can't let the remote site know that we failed to integrate their records... is that fine? How would you do this?